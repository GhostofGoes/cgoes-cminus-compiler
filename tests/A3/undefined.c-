// C-F15
main()
{
    *y;
    x=3;
    foreach(x in y);
        false | true & x < 222 + y*444;
        444*y + 222 > x & true | false;
        (444*((x) + 222) > 111) & (y | false);
    x = true | ! - ? * array * - - 42 / 13 % 5 + 31 - 331 == 12 & 7 < 9;
}

int cat()
{
    dog();
    dog = 3;
    dog = dog * dog;

    mouse = 7;

    {
        int mouse;

        mouse = 8;
    }
    mouse = 9;
    {
        mouse = 10;
    }
    mouse = 11;

    return mouse;
}
         
int dog()
{
    mouse & mouse;
    mouse > mouse;
    mouse == mouse;
    mouse += mouse;

    if (mouse) 666;
    while (mouse) 666;
    foreach (mouse in mouse) 666;
}
