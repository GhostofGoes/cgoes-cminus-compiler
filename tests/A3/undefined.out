ERROR(4): Symbol 'y' is not defined.
ERROR(5): Symbol 'x' is not defined.
ERROR(6): Symbol 'x' is not defined.
ERROR(6): Symbol 'y' is not defined.
ERROR(7): Symbol 'x' is not defined.
ERROR(7): Symbol 'y' is not defined.
ERROR(8): Symbol 'y' is not defined.
ERROR(8): Symbol 'x' is not defined.
ERROR(9): Symbol 'x' is not defined.
ERROR(9): Symbol 'y' is not defined.
ERROR(10): Symbol 'x' is not defined.
ERROR(10): Symbol 'array' is not defined.
ERROR(15): Symbol 'dog' is not defined.
ERROR(16): Symbol 'dog' is not defined.
ERROR(17): Symbol 'dog' is not defined.
ERROR(17): Symbol 'dog' is not defined.
ERROR(17): Symbol 'dog' is not defined.
ERROR(19): Symbol 'mouse' is not defined.
ERROR(26): Symbol 'mouse' is not defined.
ERROR(28): Symbol 'mouse' is not defined.
ERROR(30): Symbol 'mouse' is not defined.
ERROR(32): Symbol 'mouse' is not defined.
ERROR(37): Symbol 'mouse' is not defined.
ERROR(37): Symbol 'mouse' is not defined.
ERROR(38): Symbol 'mouse' is not defined.
ERROR(38): Symbol 'mouse' is not defined.
ERROR(39): Symbol 'mouse' is not defined.
ERROR(39): Symbol 'mouse' is not defined.
ERROR(40): Symbol 'mouse' is not defined.
ERROR(40): Symbol 'mouse' is not defined.
ERROR(42): Symbol 'mouse' is not defined.
ERROR(43): Symbol 'mouse' is not defined.
ERROR(44): Symbol 'mouse' is not defined.
ERROR(44): Symbol 'mouse' is not defined.
WARNING(35): Expecting to return type int but function 'dog' has no return statement.
Func input returns type int [line: -1]
|Sibling: 1  Func output returns type void [line: -1]
|   Child: 0  Param *dummy* int [line: -1]
|Sibling: 2  Func inputb returns type bool [line: -1]
|Sibling: 3  Func outputb returns type void [line: -1]
|   Child: 0  Param *dummy* bool [line: -1]
|Sibling: 4  Func inputc returns type char [line: -1]
|Sibling: 5  Func outputc returns type void [line: -1]
|   Child: 0  Param *dummy* char [line: -1]
|Sibling: 6  Func outnl returns type void [line: -1]
|Sibling: 7  Func main returns type void [line: 2]
|   Child: 1  Compound [line: 3]
|   |   Child: 1  Op: * Type: int [line: 4]
|   |   |   Child: 0  Id: y Type: undefined type [line: 4]
|   |   |Sibling: 1  Assign: = Type: undefined type [line: 5]
|   |   |   Child: 0  Id: x Type: undefined type [line: 5]
|   |   |   Child: 1  Const: 3 Type: int [line: 5]
|   |   |Sibling: 2  Foreach [line: 6]
|   |   |   Child: 0  Id: x Type: undefined type [line: 6]
|   |   |   Child: 1  Id: y Type: undefined type [line: 6]
|   |   |Sibling: 3  Op: | Type: bool [line: 7]
|   |   |   Child: 0  Const: false Type: bool [line: 7]
|   |   |   Child: 1  Op: & Type: bool [line: 7]
|   |   |   |   Child: 0  Const: true Type: bool [line: 7]
|   |   |   |   Child: 1  Op: < Type: bool [line: 7]
|   |   |   |   |   Child: 0  Id: x Type: undefined type [line: 7]
|   |   |   |   |   Child: 1  Op: + Type: int [line: 7]
|   |   |   |   |   |   Child: 0  Const: 222 Type: int [line: 7]
|   |   |   |   |   |   Child: 1  Op: * Type: int [line: 7]
|   |   |   |   |   |   |   Child: 0  Id: y Type: undefined type [line: 7]
|   |   |   |   |   |   |   Child: 1  Const: 444 Type: int [line: 7]
|   |   |Sibling: 4  Op: | Type: bool [line: 8]
|   |   |   Child: 0  Op: & Type: bool [line: 8]
|   |   |   |   Child: 0  Op: > Type: bool [line: 8]
|   |   |   |   |   Child: 0  Op: + Type: int [line: 8]
|   |   |   |   |   |   Child: 0  Op: * Type: int [line: 8]
|   |   |   |   |   |   |   Child: 0  Const: 444 Type: int [line: 8]
|   |   |   |   |   |   |   Child: 1  Id: y Type: undefined type [line: 8]
|   |   |   |   |   |   Child: 1  Const: 222 Type: int [line: 8]
|   |   |   |   |   Child: 1  Id: x Type: undefined type [line: 8]
|   |   |   |   Child: 1  Const: true Type: bool [line: 8]
|   |   |   Child: 1  Const: false Type: bool [line: 8]
|   |   |Sibling: 5  Op: & Type: bool [line: 9]
|   |   |   Child: 0  Op: > Type: bool [line: 9]
|   |   |   |   Child: 0  Op: * Type: int [line: 9]
|   |   |   |   |   Child: 0  Const: 444 Type: int [line: 9]
|   |   |   |   |   Child: 1  Op: + Type: int [line: 9]
|   |   |   |   |   |   Child: 0  Id: x Type: undefined type [line: 9]
|   |   |   |   |   |   Child: 1  Const: 222 Type: int [line: 9]
|   |   |   |   Child: 1  Const: 111 Type: int [line: 9]
|   |   |   Child: 1  Op: | Type: bool [line: 9]
|   |   |   |   Child: 0  Id: y Type: undefined type [line: 9]
|   |   |   |   Child: 1  Const: false Type: bool [line: 9]
|   |   |Sibling: 6  Assign: = Type: undefined type [line: 10]
|   |   |   Child: 0  Id: x Type: undefined type [line: 10]
|   |   |   Child: 1  Op: | Type: bool [line: 10]
|   |   |   |   Child: 0  Const: true Type: bool [line: 10]
|   |   |   |   Child: 1  Op: & Type: bool [line: 10]
|   |   |   |   |   Child: 0  Op: ! Type: bool [line: 10]
|   |   |   |   |   |   Child: 0  Op: == Type: bool [line: 10]
|   |   |   |   |   |   |   Child: 0  Op: - Type: int [line: 10]
|   |   |   |   |   |   |   |   Child: 0  Op: + Type: int [line: 10]
|   |   |   |   |   |   |   |   |   Child: 0  Op: % Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   Child: 0  Op: / Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   Child: 0  Op: * Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   Child: 0  Op: - Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   |   Child: 0  Op: ? Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   |   |   Child: 0  Op: * Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   |   |   |   Child: 0  Id: array Type: undefined type [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   Child: 1  Op: - Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   |   Child: 0  Op: - Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   |   |   |   Child: 0  Const: 42 Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   |   Child: 1  Const: 13 Type: int [line: 10]
|   |   |   |   |   |   |   |   |   |   Child: 1  Const: 5 Type: int [line: 10]
|   |   |   |   |   |   |   |   |   Child: 1  Const: 31 Type: int [line: 10]
|   |   |   |   |   |   |   |   Child: 1  Const: 331 Type: int [line: 10]
|   |   |   |   |   |   |   Child: 1  Const: 12 Type: int [line: 10]
|   |   |   |   |   Child: 1  Op: < Type: bool [line: 10]
|   |   |   |   |   |   Child: 0  Const: 7 Type: int [line: 10]
|   |   |   |   |   |   Child: 1  Const: 9 Type: int [line: 10]
|Sibling: 8  Func cat returns type int [line: 13]
|   Child: 1  Compound [line: 14]
|   |   Child: 1  Call: dog Type: undefined type [line: 15]
|   |   |Sibling: 1  Assign: = Type: undefined type [line: 16]
|   |   |   Child: 0  Id: dog Type: undefined type [line: 16]
|   |   |   Child: 1  Const: 3 Type: int [line: 16]
|   |   |Sibling: 2  Assign: = Type: undefined type [line: 17]
|   |   |   Child: 0  Id: dog Type: undefined type [line: 17]
|   |   |   Child: 1  Op: * Type: int [line: 17]
|   |   |   |   Child: 0  Id: dog Type: undefined type [line: 17]
|   |   |   |   Child: 1  Id: dog Type: undefined type [line: 17]
|   |   |Sibling: 3  Assign: = Type: undefined type [line: 19]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 19]
|   |   |   Child: 1  Const: 7 Type: int [line: 19]
|   |   |Sibling: 4  Compound [line: 21]
|   |   |   Child: 0  Var mouse int [line: 22]
|   |   |   Child: 1  Assign: = Type: int [line: 24]
|   |   |   |   Child: 0  Id: mouse Type: int [line: 24]
|   |   |   |   Child: 1  Const: 8 Type: int [line: 24]
|   |   |Sibling: 5  Assign: = Type: undefined type [line: 26]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 26]
|   |   |   Child: 1  Const: 9 Type: int [line: 26]
|   |   |Sibling: 6  Compound [line: 27]
|   |   |   Child: 1  Assign: = Type: undefined type [line: 28]
|   |   |   |   Child: 0  Id: mouse Type: undefined type [line: 28]
|   |   |   |   Child: 1  Const: 10 Type: int [line: 28]
|   |   |Sibling: 7  Assign: = Type: undefined type [line: 30]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 30]
|   |   |   Child: 1  Const: 11 Type: int [line: 30]
|   |   |Sibling: 8  Return [line: 32]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 32]
|Sibling: 9  Func dog returns type int [line: 35]
|   Child: 1  Compound [line: 36]
|   |   Child: 1  Op: & Type: bool [line: 37]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 37]
|   |   |   Child: 1  Id: mouse Type: undefined type [line: 37]
|   |   |Sibling: 1  Op: > Type: bool [line: 38]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 38]
|   |   |   Child: 1  Id: mouse Type: undefined type [line: 38]
|   |   |Sibling: 2  Op: == Type: bool [line: 39]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 39]
|   |   |   Child: 1  Id: mouse Type: undefined type [line: 39]
|   |   |Sibling: 3  Assign: += Type: int [line: 40]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 40]
|   |   |   Child: 1  Id: mouse Type: undefined type [line: 40]
|   |   |Sibling: 4  If [line: 42]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 42]
|   |   |   Child: 1  Const: 666 Type: int [line: 42]
|   |   |Sibling: 5  While [line: 43]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 43]
|   |   |   Child: 1  Const: 666 Type: int [line: 43]
|   |   |Sibling: 6  Foreach [line: 44]
|   |   |   Child: 0  Id: mouse Type: undefined type [line: 44]
|   |   |   Child: 1  Id: mouse Type: undefined type [line: 44]
|   |   |   Child: 2  Const: 666 Type: int [line: 44]
Number of warnings: 1
Number of errors: 34
